src/test/java/edu/ucsb/cs156/frontiers/controllers/CoursesControllerTests.java

```diff
@@ -752,6 +752,114 @@ public void delete_success_returns_ok() throws Exception {
                 assertEquals(expectedJson, responseString);
         }
+
+        @Test
+        @WithInstructorCoursePermissions
+        public void update_course_not_found_returns_not_found() throws Exception {
+                when(courseRepository.findById(eq(1L))).thenReturn(Optional.empty());
+                MvcResult response = mockMvc.perform(put("/api/courses")
+                        .param("courseId", "1")
+                        .param("courseName", "Updated Course")
+                        .param("term", "F25")
+                        .param("school", "Updated School")
+                        .with(csrf()))
+                        .andExpect(status().isNotFound())
+                        .andReturn();
+                String responseString = response.getResponse().getContentAsString();
+                Map<String, String> expectedMap = Map.of(
+                        "type", "EntityNotFoundException",
+                        "message", "Course with id 1 not found");
+                String expectedJson = mapper.writeValueAsString(expectedMap);
+                assertEquals(expectedJson, responseString);
+                verify(courseRepository).findById(eq(1L));
+                verifyNoMoreInteractions(courseRepository);
+        }
+
+        @Test
+        @WithInstructorCoursePermissions
+        public void update_course_success_returns_ok() throws Exception {
+                User user = currentUserService.getCurrentUser().getUser();
+                
+                Course originalCourse = Course.builder()
+                        .id(1L)
+                        .courseName("Original Course")
+                        .term("S25")
+                        .school("Original School")
+                        .creator(user)
+                        .build();
+                
+                Course updatedCourse = Course.builder()
+                        .id(1L)
+                        .courseName("Updated Course")
+                        .term("F25")
+                        .school("Updated School")
+                        .creator(user)
+                        .build();
+                
+                when(courseRepository.findById(eq(1L))).thenReturn(Optional.of(originalCourse));
+                when(courseRepository.save(any(Course.class))).thenReturn(updatedCourse);
+                
+                MvcResult response = mockMvc.perform(put("/api/courses")
+                        .param("courseId", "1")
+                        .param("courseName", "Updated Course")
+                        .param("term", "F25")
+                        .param("school", "Updated School")
+                        .with(csrf()))
+                        .andExpect(status().isOk())
+                        .andReturn();
+                
+                verify(courseRepository).findById(eq(1L));
+                verify(courseRepository).save(originalCourse);
```

Comment:
```suggestion
                verify(courseRepository).save(updatedCourse);
```

You should verify against the changes, rather than checking that the original (which is the same object) is correct. Please make a similar change in `admin_can_update_course_created_by_someone_else()`